Tasks/GulpV1/Tests/L0.ts

```diff
@@ -1,16 +1,292 @@
+        }
+        assert(tr.invokedToolCount == 1, 'should have only run Gulp');
+        assert(tr.stderr.length == 0, 'should not have written to stderr');
+        assert(tr.succeeded, 'task should have succeeded');
+
+        done();
+    });
+
+    it('fails if gulpFile not set', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0GulpFileNotSet.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('Input required: gulpFile'), 'Should have printed: Input required: gulpFile');
+        assert(tr.invokedToolCount == 0, 'should exit before running Gulp');
+        assert(tr.failed, 'should have failed');
+
+        done();
+    });
+
+    it('fails if cwd not set', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0CwdNotSet.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('Input required: cwd'), 'Should have printed: Input required: cwd');
+        assert(tr.invokedToolCount == 0, 'should exit before running Gulp');
+        assert(tr.failed, 'should have failed');
+
+        done();
+    });
+
+    it('runs a gulpfile through global gulp if gulpjs not set', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0GulpjsNotSet.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('gulpjs not set'), 'Should have printed: gulpjs not set');
+        assert(tr.invokedToolCount == 1, 'should have only run Gulp');
+        assert(tr.succeeded, 'task should have succeeded');
+
+        done();
+    });
+
+    it('fails if gulpFile not found', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoGulpFile.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('Not found gulpfile.js'), 'Should have printed: Not found gulpfile.js');
+        assert(tr.invokedToolCount == 0, 'should exit before running Gulp');
+        assert(tr.failed, 'should have failed');
+
+        done();
+    });
+
+    it('fails if gulp no exist globally and locally', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoGulp.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('loc_mock_GulpNotInstalled'), 'Should have printed: loc_mock_GulpNotInstalled');
+        assert(tr.failed, 'should have failed');
+
+        done();
+    });
+
+    it('fails if npm fails', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NpmFails.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('loc_mock_NpmFailed'), 'Should have printed: loc_mock_NpmFailed');
+        assert(tr.invokedToolCount == 2, 'should exit before running Gulp');
```

Comment:
Does it make sense to use `assert.equal()` here? It would possibly be easier to debug if something goes wrong.

Reply:
Thanks for the note! I'd just prefer to leave it as is for now, since grunt, gulpv0, and gulpv1 tests have the same structure. So then we can fix it there as well (if it's really necessary)